src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -220,7 +220,7 @@ m_cursorGrabbed   (m_fullscreen)
     m_handle = CreateWindowW(className, title.toWideString().c_str(), win32Style, left, top, width, height, NULL, NULL, GetModuleHandle(NULL), this);
 
     // Register to receive device interface change notifications (used for joystick connection handling)
-    DEV_BROADCAST_DEVICEINTERFACE deviceInterface = {sizeof(DEV_BROADCAST_DEVICEINTERFACE), DBT_DEVTYP_DEVICEINTERFACE, 0, GUID_DEVINTERFACE_HID, 0};
+    DEV_BROADCAST_DEVICEINTERFACE deviceInterface = {sizeof(DEV_BROADCAST_DEVICEINTERFACE), DBT_DEVTYP_DEVICEINTERFACE, 0, GUID_DEVINTERFACE_HID, {0}};
```

Comment:
Is this C++03 compliant?

For some reason I'm having a hard time to understand a) why the braces are needed and b) what it's actually constructing here... 🤔

Reply:
I think braces are required because the struct that is constructed requires not a char, but an array with a single char in it
```cpp
char  dbcc_name[1];
```
So you can't really initialize an array with a number

And the C++03 standard says that any aggregate type can be brace-initialized, so I would guess it works out

I will test it tomorrow with the 03 standard and share the results 😄

Reply:
https://godbolt.org/z/rrsca7GMY

Even ancient versions of GCC are fine with the extra braces around that `0` so I think we're good and not accidentally using modern C++ in SFML 2.